### PR TITLE
for chained references, prefer outer catalog item id, and outer tags

### PR DESCRIPTION
**Pull request by ahgittin**
as described in changes to release-notes.md

### CODE REVIEW

File: brooklyn-server/api/src/main/java/org/apache/brooklyn/api/internal/AbstractBrooklynObjectSpec.java [hunk 1]
```diff
@@ -105,9 +118,19 @@ public SpecT tag(Object tag) {
...
     /** adds the given tags */
     public SpecT tags(Iterable<Object> tagsToAdd) {
+        return tagsAdd(tagsToAdd);
+    }
+    /** adds the given tags */
+    public SpecT tagsAdd(Iterable<Object> tagsToAdd) {
         Iterables.addAll(this.tags, tagsToAdd);
         return self();
     }
+    /** replaces the given tags */
```
Comment (aledsage) [inline brooklyn-server/api/src/main/java/org/apache/brooklyn/api/internal/AbstractBrooklynObjectSpec.java]: Changes description to "replaces the existing tags"?

File: brooklyn-server/core/src/main/java/org/apache/brooklyn/core/typereg/AbstractTypePlanTransformer.java [hunk 2]
```diff
@@ -103,7 +103,8 @@ public Object create(final RegisteredType type, final RegisteredTypeLoadingConte
                 @Override protected Object visitSpec() {
                     try { 
                         AbstractBrooklynObjectSpec<?, ?> result = createSpec(type, context);
-                        result.catalogItemId(type.getId());
+                        // see notes in CampResolver setting catalogItemId; we might need to keep both
```
Comment (aledsage) [inline brooklyn-server/core/src/main/java/org/apache/brooklyn/core/typereg/AbstractTypePlanTransformer.java]: I do see any new notes in `CampResolver` - did you mean in `AbstractBrooklynObjectSpec`?

**Comment by aledsage** [comment]
With a quick look, seems reasonable. Only one very minor comment - do you want to look at that and then merge @ahgittin ?

**Comment by ahgittin** [comment]
tidied as suggested, merging

**Pull request merged**